File: sdks/packages/midan/src/events-types/update-event-team-event.ts
import { PublicKey } from '@solana/web3.js';
import { BN } from '@coral-xyz/anchor';
import {EventTeamStatusType} from "../types/midan-enums";

export const UpdateEventTeamEventName = "UpdateEventTeamEvent";

export interface UpdateEventTeamEvent {
    timestamp: BN,

    creatorKey: PublicKey,

    authority: PublicKey,

    eventTeamIndex: number,

    eventTeamStatus: EventTeamStatusType
}

File: programs/midan/src/events/update_event_team_event.rs
use anchor_lang::prelude::*;
use crate::states::EventTeamStatus;

#[event]
pub struct UpdateEventTeamEvent {
    pub timestamp: i64,

    pub creator_key: Pubkey,

    pub authority: Pubkey,

    pub event_team_index: u32,
    
    pub event_team_status: EventTeamStatus,
}